starlark/value.go

```diff
@@ -377,10 +377,17 @@ func (x Bool) CompareSameType(op syntax.Token, y_ Value, depth int) (bool, error
 // Float is the type of a Starlark float.
 type Float float64
 
-func (f Float) String() string { return strconv.FormatFloat(float64(f), 'g', 6, 64) }
-func (f Float) Type() string   { return "float" }
-func (f Float) Freeze()        {} // immutable
-func (f Float) Truth() Bool    { return f != 0.0 }
+func (f Float) String() string {
+	s := strconv.FormatFloat(float64(f), 'g', 6, 64)
+	// See https://github.com/bazelbuild/starlark/issues/130.
+	if !strings.ContainsAny(s, ".eNI") { // 1.5 is ok, 1e5 is ok, NaN is ok, Inf is ok
+		s += ".0"
```

Comment:
Neat!  However this adds an allocation in the common case. Use a byte slice and strconv.AppendFloat followed by append(".0")? Alternatively handle the common cases where float64(int64(f))==f separately?